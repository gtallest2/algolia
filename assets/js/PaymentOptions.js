import React from 'react'

const { func, bool } = React.PropTypes

const PaymentOptions = React.createClass({
  propTypes: {
    mobileMenu: bool,
    handlePaymentToggle: func
  },

  handlePaymentToggle (cardType) {
    this.props.handlePaymentToggle(cardType)
  },

  render () {
    return (
      <div className='payments-filter'>
        <h4>Payment Options</h4>
        <ul className='payment-options'>
          <li className='payment-type'>
            <input onClick={this.handlePaymentToggle.bind(null, 'AMEX')} type='checkbox' />
            <span className='card cc-ae' />&nbsp;
            <span className='amex-card-name'>{this.props.mobileMenu ? 'AMEX' : 'American Express'}</span>
          </li>
          <li className='payment-type'>
            <input onClick={this.handlePaymentToggle.bind(null, 'Discover')} type='checkbox' />
            <span className='card cc-discover' />&nbsp;
            <span className='discover-card-name'>Discover</span>
          </li>
          <li className='payment-type'>
            <input onClick={this.handlePaymentToggle.bind(null, 'MasterCard')} type='checkbox' />
            <span className='card cc-mc' />&nbsp;
            <span className='mastercard-card-name'>MasterCard</span>
          </li>
          <li className='payment-type'>
            <input onClick={this.handlePaymentToggle.bind(null, 'Visa')} type='checkbox' />
            <span className='card cc-visa' />&nbsp;
            <span className='visa-card-name'>Visa</span>
          </li>
        </ul>
      </div>
    )
  }
})

export default PaymentOptions
