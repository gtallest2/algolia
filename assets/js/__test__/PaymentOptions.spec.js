import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import PaymentOptions from '../PaymentOptions'

test('PaymentOptions snapshot test', () => {
  const component = shallow(<PaymentOptions />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})


test('PaymentOptions displays "AMEX" instead of "American Express" on mobile', () => {
  const component = shallow(<PaymentOptions mobileMenu={true} />)
  expect(component.find('.amex-card-name').text()).toEqual('AMEX');
})
