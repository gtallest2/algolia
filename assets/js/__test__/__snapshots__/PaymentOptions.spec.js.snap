exports[`test PaymentOptions snapshot test 1`] = `
<div
  className="payments-filter">
  <h4>
    Payment Options
  </h4>
  <ul
    className="payment-options">
    <li
      className="payment-type">
      <input
        onClick={[Function]}
        type="checkbox" />
      <span
        className="card cc-ae" />
       
      <span
        className="amex-card-name">
        American Express
      </span>
    </li>
    <li
      className="payment-type">
      <input
        onClick={[Function]}
        type="checkbox" />
      <span
        className="card cc-discover" />
       
      <span
        className="discover-card-name">
        Discover
      </span>
    </li>
    <li
      className="payment-type">
      <input
        onClick={[Function]}
        type="checkbox" />
      <span
        className="card cc-mc" />
       
      <span
        className="mastercard-card-name">
        MasterCard
      </span>
    </li>
    <li
      className="payment-type">
      <input
        onClick={[Function]}
        type="checkbox" />
      <span
        className="card cc-visa" />
       
      <span
        className="visa-card-name">
        Visa
      </span>
    </li>
  </ul>
</div>
`;
